syntax = "proto3";
package mongo;




import "google/protobuf/struct.proto";

/*
SearchOne请求
*/
message SearchOneRequest {
    /*
    表名
    */
    string collection = 1 ;
    /*
    查询条件
    */
    google.protobuf.Struct query = 2 ;
    /*
    fields
    */
    repeated string fields = 3 ;
    /*
    sort
    */
    repeated string sort = 4 ;
}

/*
SearchOne返回
*/
message SearchOneResponse {
    /*
    记录
    */
    google.protobuf.Struct doc = 1 ;
}

/*
SearchOneApi返回
*/
message SearchOneResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    mongo.SearchOneResponse data = 4 ;
}

